lib/workers/repository/process/index.ts

```diff
@@ -56,6 +56,14 @@ async function getBaseBranchConfig(
       throw error;
     }
 
+    // istanbul ignore if
+    if (config.printConfig) {
+      logger.info(
+        { config: baseBranchConfig },
+        'BaseBranchConfig overrides all configurations'
+      );
+    }
+
     baseBranchConfig = mergeChildConfig(config, baseBranchConfig);
```

Comment:
@TWiStErRob did you intend for this new log to print the base branch config before or after it is merged on top of the repo config?

Reply:
but there is no way to do that efficiently @rarkins,
the only way to do that, is if we print the WHOLE config for each branch... filling logs with prints for each branch which come late. this config print comes just before the extraction phase,

i think adding print config after each branch for what changed, could be the best indicator as it is now, 
as @TWiStErRob mentioned that there are no logs showing the field was overridden, and we dont print the branch config anytime.

Reply:
Doesn't printConfig already print the _whole_ config?

Reply:
It prints the config before it gets updated with branch config. Look at the logs. The whole task is about branch config not showing in the logs

Reply:
That wasn't the question I was asking. The question is whether we should print either:
- Just the raw config in the file in the branch, or
- The full branch config after that file has been merged against the main config

I believe that printConfig=true prints the _full_ config, so it would be consistent to do this for the branch too.

Reply:
Thats what i meant when i said that it will fill the logs for each branch...
I just wanted the flat before merge si it wouldnt spam configs it will get confusing

Reply:
<img width="951" alt="image" src="https://user-images.githubusercontent.com/6311784/202469408-c452d5db-dd1e-4cb6-abf3-1763075c8c68.png">

`printConfig` already prints the fully resolved config for the default branch, it should do the same for any non-default baseBranch too. And if we don't already, we should first log the small/raw baseBranch config prior to the fully resolved/merged one

Reply:
> lready prints the fully resolved config for the default branch, it should do the same for any non-default baseBranch too. and if we don't already

no we do not print basebranch config at all currently, we print only the branch itself.
```
    logger.debug(
      { baseBranch },
      `Merging config from base branch because useBaseBranchConfig=merge`
    );
```

>we should first log the small/raw baseBranch config prior to the fully resolved/merged one

`prior` is not possible because the basebranch config is merged after the print => 
Current print exists at `workers/repository/init/index.ts` :: `initRepo()` 

Then if you look a the issue comment code that i mentioned. https://github.com/renovatebot/renovate/issues/17635#issuecomment-1305409925


in the `Extract` + `Lookup` sections for each branch, we do `getBaseBranchConfig()` which will merge base branch config on top of the current config ( overriding it)

so the options are :
1. it will print before the extraction, and before the lookup ( raw config twice ) like i did. or  
2. print the fully merged config for each base branch before extract starts, right after 
`const baseBranchConfig = await getBaseBranchConfig(baseBranch, config);`

the current implementation is option 1. because i didnt wanna spam the logs with full configs.

Reply:
why would we call `await getBaseBranchConfig(baseBranch, config);` twice? before extract and before lookups? will it change after extract?

Reply:
I think you can do this:

```
    logger.debug(
      { baseBranch, config: baseBranchConfig },
      `Merging config from base branch because useBaseBranchConfig=merge`
    );
```

And then add another statement which prints the full config after the merge happens. It's not spamming the logs if users have enabled printConfig=true, because it's purpose is to print the _full_ logs.

Reply:
@rarkins FYI the baseBranchConfig is not a small simple config, its as large as the main config,

So basically what im doing is 
1) print all the base branch config
2) print all base branch config after merging with actual config.

i find it that printing the small Delta is not possible, 
on the other hand i could skip step 1, and print the merged base branch config and that's enough imo

Reply:
i changed it as i see best, see uploaded example:
[printconfig.log](https://github.com/renovatebot/renovate/files/10049675/printconfig.log)